src/product/ProductModel.js

```diff
@@ -1,13 +1,28 @@
-import { types } from 'mobx-state-tree'
+import { types, flow } from 'mobx-state-tree'
 import ProductModelBase from 'react-storefront/model/ProductModelBase'
 
 const ProductModel = types.compose(
   ProductModelBase,
-  types.model('ProductModel', {
-    // additional product fields go here
-    specs: types.maybeNull(types.string),
-    reviews: types.optional(types.array(types.string), [])
-  })
+  types
+    .model('ProductModel', {
+      // additional product fields go here
+      specs: types.maybeNull(types.string),
+      reviews: types.optional(types.array(types.string), []),
+      recommendations: types.maybeNull(types.array(types.late(() => ProductModel)))
+    })
+    .actions(self => ({
+      /**
+       * Late loads personalization data from product/personalization-handler.js
+       */
+      loadPersonalization: flow(function*() {
```

Comment:
Is it required to use generators here? I think we should stay as simple as possible in the boilerplate code.

Reply:
Could we use:
```js
self.recommendations = await fetch(`/p/${self.id}/personalization.json`)
  .then(res => res.json())
  .then(json => json.recommendations)

```